### PR TITLE
TS-4449 header_rewrite: Better errors and debug output

### PR DESCRIPTION
**Pull request by zwoop**

### CODE REVIEW

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/331/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/225/ for details.

File: plugins/header_rewrite/conditions.cc [hunk 1]
```diff
@@ -86,9 +86,8 @@ ConditionMethod::eval(const Resources &res)
   std::string s;
 
   append_value(s, res);
-  bool rval = static_cast<const Matchers<std::string> *>(_matcher)->test(s);
-  TSDebug(PLUGIN_NAME, "Evaluating METHOD(): %s - rval: %d", s.c_str(), rval);
-  return rval;
+  TSDebug(PLUGIN_NAME, "Evaluating METHOD()");
```
Comment (jpeach) [inline plugins/header_rewrite/conditions.cc]: Why did you drop the value from this debug message? The one below in ``ConditionHeader`` still has it ...

File: plugins/header_rewrite/conditions.cc [hunk 2]
```diff
@@ -316,12 +314,12 @@ ConditionQuery::eval(const Resources &res)
 {
   std::string s;
 
+  TSDebug(PLUGIN_NAME, "Evaluating QUERY()");
   if (NULL == res._rri) {
-    TSDebug(PLUGIN_NAME, "QUERY requires remap initialization! Evaluating to false!");
+    TSDebug(PLUGIN_NAME, "\tQUERY requires remap initialization! Evaluating to false!");
```
Comment (jpeach) [inline plugins/header_rewrite/conditions.cc]: This seems like a configuration problem. Maybe it should be a ``TSError``.

File: plugins/header_rewrite/conditions.cc [hunk 3]
```diff
@@ -584,16 +585,17 @@ bool
 ConditionIncomingPort::eval(const Resources &res)
 {
   uint16_t port = getPort(TSHttpTxnIncomingAddrGet(res.txnp));
-  bool rval     = static_cast<const Matchers<uint16_t> *>(_matcher)->test(port);
-  TSDebug(PLUGIN_NAME, "Evaluating INCOMING-PORT(): %d: rval: %d", port, rval);
-  return rval;
+
+  TSDebug(PLUGIN_NAME, "Evaluating INCOMING-PORT()");
```
Comment (jpeach) [inline plugins/header_rewrite/conditions.cc]: Print the port?

File: plugins/header_rewrite/regex_helper.h [hunk 4]
```diff
@@ -36,11 +36,13 @@ class regexHelper
...
   {
-    if (regex)
+    if (regex) {
       pcre_free(regex);
+    }
 
-    if (regexExtra)
+    if (regexExtra) {
       pcre_free(regexExtra);
+    }
```
Comment (jpeach) [inline plugins/header_rewrite/regex_helper.h]: FWIW, ``pcre_free`` should have the same semantics as ``free(3)`` so passing ``NULL`` is a no-op.

**Comment by jpeach** [comment]
Looks good; I just had a couple of trivial suggestions.

👍

File: plugins/header_rewrite/conditions.cc (hunk 3 above)
Comment (zwoop) [inline plugins/header_rewrite/conditions.cc]: I think it prints that as part of the other code, when doing the matching.

File: plugins/header_rewrite/conditions.cc (hunk 2 above)
Comment (zwoop) [inline plugins/header_rewrite/conditions.cc]: Maybe, but that might be a different bug to fix honestly. File a Jira? We shouldn't error out here I don't think.

**Comment by atsci** [comment]
Linux build *failed*! See https://ci.trafficserver.apache.org/job/Github-Linux/239/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/345/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/240/ for details.

**Comment by zwoop** [comment]
I cleaned up a few of the missing pieces, made a get() method better (less STL copy on use), and added some typedef's as introduced in one of the operators (it makes sense and cleans up the usage in other places).

**Pull request closed without merging**